eng/build-job.yml

```diff
@@ -174,12 +127,12 @@ jobs:
       - script: set __TestIntermediateDir=int&&build.cmd $(buildConfig) $(archType) -ci -skiptests -skipbuildpackages $(officialBuildIdArg) $(ibcOptimizeArg) $(enforcePgoArg)
         displayName: Build product
 
-    # Build native test components
+    # Build native test components and crossgen the framework assemblies
     - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
-      - script: ./build-test.sh skipmanaged $(buildConfig) $(archType) $(crossArg) $(priorityArg) $(clangArg)
+      - script: ./build-test.sh skipmanaged crossgen $(buildConfig) $(archType) $(crossArg) $(priorityArg) $(clangArg)
```

Comment:
Do we always want to crossgen the framework assemblies for testing?

Reply:
Good point. I'll add a jobParameter readyToRun to build-job so that the individual pipelines can choose whether they want to crossgen the framework (i.o.w. whether they intend to run R2R test jobs).

Reply:
Fixed in 5th commit.